filesystems/s3/src/main/scala/cromwell/filesystems/s3/batch/S3BatchIoCommand.scala

```diff
@@ -93,22 +93,39 @@ sealed trait S3BatchHeadCommand[T] extends S3BatchIoCommand[T, HeadObjectRespons
   def file: S3Path
 }
 
+/**
+  * `IoCommand` to find the size (content length) of an S3 object
+  * @param file the path to the object
+  */
 case class S3BatchSizeCommand(override val file: S3Path) extends IoSizeCommand(file) with S3BatchHeadCommand[Long] {
   override def mapResponse(response: HeadObjectResponse): Long = response.contentLength
 }
 
+/**
+  * `IoCommand` to find the hash of an s3 object (the `Etag`)
+  * @param file the path to the object
+  */
 case class S3BatchEtagCommand(override val file: S3Path) extends IoHashCommand(file) with S3BatchHeadCommand[String] {
   override def mapResponse(response: HeadObjectResponse): String = response.eTag
 }
 
+/**
+  * `IoCommand` to "touch" an S3 object. The current implementation of `mapResponse` in this object doesn't do anything
+  * as it is not clear that touch is meaningful in the context of S3
```

Comment:
You can create a zero size object in s3 but if you touch an existing file you wouldn't want to create a zero size object.  In some ways this gets to the thorny issue of cromwell expecting a file system and object stores faking it rather than expecting an object store and making a file system look like one if you have an actual file system.